pkg/athena/api/api.go

```diff
@@ -73,6 +73,11 @@ func (c *API) Execute(ctx context.Context, input *api.ExecuteQueryInput) (*api.E
 	return &api.ExecuteQueryOutput{ID: *output.QueryExecutionId}, nil
 }
 
+// GetQueryID returns not found, as calling ListQueryExecutions gets throttled when there are many panels
```

Comment:
Hey sorry I'm not sure I totally understand what this comment/function does, do you mind talking through it a bit more?

Reply:
This used to use the [`ListQueryExecutions` API](https://docs.aws.amazon.com/athena/latest/APIReference/API_ListQueryExecutions.html) to get a list of query ids and then call [`BatchGetQueryExecutions`](https://docs.aws.amazon.com/athena/latest/APIReference/API_BatchGetQueryExecution.html) to get the current status of a query and to be able to match the currently running query string.

The limit for `ListQueryExecutions` is 5 calls per second https://docs.aws.amazon.com/athena/latest/ug/service-limits.html#service-limits-api-calls so it gets exhausted quite quickly with multiple panels and if we need to page through the results of the call to find the matching query id for the current running query.

Reply:
Rewrote the comment to be clearer.

Reply:
rad makes sense thanks!!